File: Backend/src/main/java/com/NoHexRoger/Backend/Entity/Estancia.java
package com.NoHexRoger.Backend.Entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;


@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
@Entity(name = "estancia")
public class Estancia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private LocalDateTime fechaEntrada;

    @Column
    private LocalDateTime fechaSalida;

    @ManyToOne
    @JoinColumn(name = "vehiculo_id", nullable = false)
    private Vehiculo vehiculo;

}

File: Frontend/src/app/shared/estancia.ts
export interface Estancia {
  id: number;
  fechaEntrada: string;
  fechaSalida?: string;
  vehiculo: {
    placa: string;
    minutosAcumulados: number;
    deudaAcumulada: number;
    tipoVehiculo: {
      id: number;
      tipo: string;
    };
  };
}
